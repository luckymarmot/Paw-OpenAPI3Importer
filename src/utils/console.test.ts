import logger from './console'

describe('logger utility', () => {
  it('should log string', () => {
    const _logger = jest.spyOn(logger, 'info')
    logger.info('Hello,', 'How are you?')
    expect(_logger).toHaveBeenCalledWith('Hello,', 'How are you?')
    expect(_logger).toMatchSnapshot()
  })

  it('should log object and nested object', () => {
    const _logger = jest.spyOn(logger, 'info')
    const object = {
      a: 'hello',
      b: { c: { d: 'world' } },
      c: [1, 2, 3, 4],
    }
    logger.info(object, object.c)
    expect(_logger).toHaveBeenCalledWith(object, object.c)
    expect(_logger).toMatchSnapshot()
  })
})
