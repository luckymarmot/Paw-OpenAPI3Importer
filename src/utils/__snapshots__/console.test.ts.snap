// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logger utility should log object and nested object 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      Object {
        "a": "hello",
        "b": Object {
          "c": Object {
            "d": "world",
          },
        },
        "c": Array [
          1,
          2,
          3,
          4,
        ],
      },
      Array [
        1,
        2,
        3,
        4,
      ],
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`logger utility should log string 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "Hello,",
      "How are you?",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
